nifi-asymmetrik-standard-bundle/nifi-asymmetrik-standard-reporting-tasks/src/main/java/com/asymmetrik/nifi/reporting/AbstractNiFiClusterMetricsReporter.java

```diff
@@ -376,26 +379,21 @@ private void populateOutputPortSnapshots(ProcessGroupStatus status, SystemMetric
         return metrics;
     }
 
-    /**
-     * Converts IP address specified as byte[] to dot notation String equivalent
-     *
-     * @return String representation of ipAddress
-     */
-    public static String getIpAddress() {
```

Comment:
Are there other metrics reporters that might need the ip address?

Reply:
yes, the influx reporter uses the IP address.  Since these are AWS-specific changes, none of these changes should be in the abstract class. It may be better to subclass this abstract class with another AWS-specific abstract class.

As a concrete example, I run an instance of NiFi and Influx on my development machine and use the Influx reporting task.  With the changes proposed here, this would not be possible because the Abstract reporting task depends on AWS tags.  There are other users that may want to run on Azure or GCP and they would not be able to use any of our reporting tasks.